### ISSUE TITLE
Metadata templates

### ISSUE THREAD

**Issue by michael-conway** [DICE-UNC/irods-cloud-browser#131]
Per discussion, implement metadata templates as a tabbed view

**Comment by michael-conway** [comment]
depends on #132

**Comment by burnce** [comment]
I am assuming this feature means that the browser can be configured to request a predefined set of metadata from the user on file upload into iRODS.
Will values filled automatically by policies defined in the rule engine be prefilled?

**Comment by michael-conway** [comment]
Yes, this was Reagan's idea. Although for this first iteration we are 
only focusing on user entry, that is, the ability to dynamically 
generate data entry forms and validation for metadata templates.

Reagan had outlined the idea of metadata templates also able to specify 
'automatic' metadata, and this is incorporated into the design.

The idea is that a template describes a set of elements.  Each element 
can specify a procedure (rule) that derives that metadata value, and a 
mapping of an output parameter of that procedure that is marshaled back 
into an AVU.

This is neat, because you can dynamically apply metadata extraction 
rules by merit of applying a metadata template.
